File: src/main/scala/fi/finlit/edith/ui/pages/NoteImport.scala
package fi.finlit.edith.ui.pages

import java.io.File
import java.io.IOException
import org.apache.tapestry5.PersistenceConstants
import org.apache.tapestry5.annotations.Persist
import org.apache.tapestry5.annotations.Property
import org.apache.tapestry5.ioc.Messages
import org.apache.tapestry5.ioc.annotations.Inject
import org.apache.tapestry5.upload.services.UploadedFile
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import fi.finlit.edith.ui.services.NoteDao
import NoteImport._
//remove if not needed
import scala.collection.JavaConversions._

object NoteImport {

  private val logger = LoggerFactory.getLogger(classOf[NoteImport])
}

class NoteImport {

  @Inject
  private var noteDao: NoteDao = _

  @Property
  private var file: UploadedFile = _

  @Inject
  private var messages: Messages = _

  @Persist(PersistenceConstants.FLASH)
  @Property
  private var message: String = _

  def onActivate() {
  }

  def onSuccess() {
    var tempFile = File.createTempFile("upload", null)
    try {
      file.write(tempFile)
      var rv = noteDao.importNotes(tempFile)
      message = messages.format("notes-imported-msg", rv)
    } finally {
      if (!tempFile.delete() && !tempFile.delete()) {
        logger.error("Delete of " + tempFile.getAbsolutePath + " failed")
      }
    }
  }
}

File: src/main/resources/com/mysema/edith/ui/pages/NoteImport.java
/*
 * Copyright (c) 2009 Mysema Ltd.
 * All rights reserved.
 *
 */
package com.mysema.edith.ui.pages;

import java.io.File;
import java.io.IOException;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.mysema.edith.services.NoteDao;
import com.sun.xml.internal.ws.api.PropertySet.Property;

@SuppressWarnings("unused")
public class NoteImport {

    private static final Logger logger = LoggerFactory.getLogger(NoteImport.class);

    @Inject
    private NoteDao noteDao;

    @Property
    private UploadedFile file;

    @Inject
    private Messages messages;

    @Persist(PersistenceConstants.FLASH)
    @Property
    private String message;

    void onActivate() {
    }

    public void onSuccess() throws IOException {
        File tempFile = File.createTempFile("upload", null);
        try {
            file.write(tempFile);
            int rv = noteDao.importNotes(tempFile);
            message = messages.format("notes-imported-msg", rv);
        } finally {
            if (!tempFile.delete() && !tempFile.delete()) {
                logger.error("Delete of " + tempFile.getAbsolutePath() + " failed");
            }
        }
    }

}
